Python/bltinmodule.c

```diff
@@ -2444,36 +2437,33 @@ builtin_sorted(PyObject *self, PyObject *const *args, Py_ssize_t nargs, PyObject
 }
 
 
-/*[clinic input]
-vars as builtin_vars
-
-    object: object = NULL
-    /
-
-Show vars.
-
-Without arguments, equivalent to locals().
-With an argument, equivalent to object.__dict__.
-[clinic start generated code]*/
-
+/* AC: cannot convert yet, as needs PEP 457 group support in inspect */
 static PyObject *
-builtin_vars_impl(PyObject *module, PyObject *object)
-/*[clinic end generated code: output=840a7f64007a3e0a input=80cbdef9182c4ba3]*/
+builtin_vars(PyObject *self, PyObject *args)
 {
+    PyObject *v = NULL;
     PyObject *d;
 
-    if (object == NULL) {
-        d = _PyEval_GetFrameLocals();
+    if (!PyArg_UnpackTuple(args, "vars", 0, 1, &v))
+        return NULL;
+    if (v == NULL) {
+        d = Py_XNewRef(PyEval_GetLocals());
```

Comment:
This should use `_PyEval_GetFrameLocals` (I think that's what's causing the test failures; maybe there's similar issues in `dir()` and `locals()`).

Reply:
Thanks, I've missed it.